### DISCUSSION TITLE
How to generate Quarkus client from OpenAPI spec?

### DISCUSSION THREAD

**Discussion by dhoffer** [quarkiverse/quarkus-openapi-generator#815] [Q&A]
How to generate Quarkus client from OpenAPI spec?  I have followed the online documention.  Placed openapi file in main/openapi/folder.

Added these dependencies

`<dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
        </dependency>

        <dependency>
            <groupId>io.quarkiverse.openapi.generator</groupId>
            <artifactId>quarkus-openapi-generator</artifactId>
            <version>2.4.7</version>
        </dependency>`

And then modified the quarkus-maven-plugin

`<goals>
                            <goal>build</goal>
                            <goal>generate-code</goal>
                            <goal>generate-code-tests</goal>
                        </goals>`

But when I build it generates empty generated-sources folders.  How to get it to generate the client code?

Also if I were to use openapi-generator-maven-plugin directly instead of quarkus-openapi-generator what library option should I use that will generate client code that works with Quarkus existing dependencies?  We are using 3.15.1 with the reactive REST JAX-RS with JDK21 so jakarta package names.

**Comment by ricardozanini** [comment]
Based on your description, everything seems fine. Can you share a reproducer?


> Also if I were to use openapi-generator-maven-plugin directly instead of quarkus-openapi-generator what library option should I use that will generate client code that works with Quarkus existing dependencies? We are using 3.15.1 with the reactive REST JAX-RS with JDK21 so jakarta package names.

I don't have this information, but at the time we created this extension, it was because the original generator did not support Quarkus.
